domain/src/main/java/com/distillery/android/domain/FakeToDoRepository.kt

```diff
@@ -0,0 +1,91 @@
+package com.distillery.android.domain
+
+import com.distillery.android.domain.models.ToDoModel
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import java.util.Date
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
+
+private val uniqueIdGenerator = AtomicLong(0)
+
+private const val DEFAULT_DELAY_GENERATOR = 5000L
+private const val DEFAULT_DELAY_ADDITION = 1500L
+private const val TIME_TO_DIE = 10L
+
+class FakeToDoRepository : ToDoRepository {
+
+    private val toDos = ConcurrentHashMap<Long, ToDoModel>()
+    private val todosChannel = Channel<List<ToDoModel>>()
+
+    init {
+        GlobalScope.launch {
+            generateValues().collect { model ->
+                toDos[model.uniqueId] = model
+                println("${ToDoRepository::class.simpleName} Publishing value ${model.uniqueId}")
+                todosChannel.send(toDos.values.toList())
+            }
+        }
+    }
+
+    override fun fetchToDos(): Flow<List<ToDoModel>> {
+        return todosChannel.receiveAsFlow()
+    }
+
+    override suspend fun addToDo(title: String, description: String) {
+        println("${ToDoRepository::class.simpleName} Started value addition")
+        delay(DEFAULT_DELAY_ADDITION)
+        val id = uniqueIdGenerator.getAndIncrement()
+        val model = ToDoModel(id, "$title ID: $id", description)
+        toDos[model.uniqueId] = model
+        println("${ToDoRepository::class.simpleName} Value ${model.uniqueId} added")
+        if (uniqueIdGenerator.get() == TIME_TO_DIE) {
+            // moment of surprise
+            println("${ToDoRepository::class.simpleName} Oops")
+            throw IllegalArgumentException("You died")
+        } else {
+            println("${ToDoRepository::class.simpleName} Publishing value ${model.uniqueId}")
+            todosChannel.send(toDos.values.toList())
+        }
+    }
+
+    override fun completeToDo(uniqueId: Long) {
+        if (uniqueId > 2 && uniqueId % 2 == 0L) {
```

Comment:
I assume you wanted to make the first completed task not throwing an exception. If so, there is a way to make the intention more explicit:
```suggestion
        if (uniqueId != 0 && uniqueId % 2 == 0L) {
```

Reply:
@0neel in that case I want it to throw exception only for 4th element and further, so we have a bit more space to manoeuvre, the idea is not to throw an exception for a particular reason, but to simulate the feel of unexpected exceptions, while having these controlled